### PR TITLE
Input: move to release candidate

### LINKED ISSUE #22674: Input: Move to RC
- [x] Resolve `contentBefore/After` Storybook stories - #21450
- [x] Resolve filledDarker, filledLighter contrast issue - #21449
- [x] Finish a11y testing - #21174
- [x] Write migration guide - #22769

### PR DESCRIPTION
**Pull request by spmonahan**
## Current Behavior

`Input` is a beta component

## New Behavior

`Input` is a release candidate component

## Related Issue(s)

Fixes #22674

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |         Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> |  `180.2 kB`<br />`50.239 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            | `34.078 kB`<br />`11.098 kB` |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/Input.fixture.js'>Input</abbr>                                                                                       |  `22.174 kB`<br />`7.304 kB` |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButton.fixture.js'>SpinButton</abbr>                                                                        | `41.865 kB`<br />`11.861 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/6cb084a7afe7d662dc2d9d65b91474f132566390'>6cb084a7afe7d662dc2d9d65b91474f132566390</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23090/perf-test-react-components/Avatar-mount_base.html">1096</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23090/perf-test-react-components/Avatar-mount.html">1076</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23090/perf-test-react-components/Button-mount_base.html">693</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23090/perf-test-react-components/Button-mount.html">680</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23090/perf-test-react-components/FluentProvider-mount_base.html">2185</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23090/perf-test-react-components/FluentProvider-mount.html">2211</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23090/perf-test-react-components/FluentProviderWithTheme-mount_base.html">314</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23090/perf-test-react-components/FluentProviderWithTheme-mount.html">355</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23090/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">282</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23090/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">266</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23090/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">377</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23090/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">345</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23090/perf-test-react-components/MakeStyles-mount_base.html">1880</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23090/perf-test-react-components/MakeStyles-mount.html">1827</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: change/@fluentui-react-components-fde1774d-c84a-4853-a18b-abce2af7196e.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "update react-input to release candidate",
```
Comment (sopranopillow) [inline change/@fluentui-react-components-fde1774d-c84a-4853-a18b-abce2af7196e.json:3]: nit:
```suggestion
  "comment": "Update react-input to release candidate.",
```

File: change/@fluentui-react-input-bed71aa5-4608-4ce3-9537-40a32dd44e69.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "move from beta to release candidate",
```
Comment (sopranopillow) [inline change/@fluentui-react-input-bed71aa5-4608-4ce3-9537-40a32dd44e69.json:3]: nit:
```suggestion
  "comment": "Move from beta to release candidate.",
```

File: change/@fluentui-react-spinbutton-c26351d8-d806-445e-939e-3b84c6e59bbb.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "update react-input dependency version to release candidate",
```
Comment (sopranopillow) [inline change/@fluentui-react-spinbutton-c26351d8-d806-445e-939e-3b84c6e59bbb.json:3]: nit:
```suggestion
  "comment": "Update react-input dependency version to release candidate.",
```

**Review by sopranopillow — APPROVED** [review]

**Review by khmakoto — APPROVED** [review]

**Pull request merged**